pkg/provider/azure.go

```diff
@@ -610,7 +613,12 @@ func (az *Cloud) InitializeCloudFromConfig(config *Config, fromSecret, callFromC
 		az.MaximumLoadBalancerRuleCount = consts.MaximumLoadBalancerRuleCount
 	}
 
-	if strings.EqualFold(consts.VMTypeVMSS, az.Config.VMType) {
+	if strings.EqualFold(consts.VMTypeVmssFlex, az.Config.VMType) {
+		az.VMSet, err = newFlexScaleSet(az)
```

Comment:
let's create a new `VMSSFlexSet` here for vmss flex when DisableVMSSFlexNodes==false?

Reply:
then in the following controllers, you could invoke the VMSet implementation based on the real node type.